### PR TITLE
Add pagination in media modal

### PR DESCRIPTION
**Pull request by pfm-eyesightjp**
modal_dot

### CODE REVIEW

File: app/javascript/mastodon/features/ui/components/media_modal.js [hunk 1]
```diff
@@ -66,13 +71,17 @@ export default class MediaModal extends ImmutablePureComponent {
 
     const index = this.getIndex();
 
-    let leftNav, rightNav, content;
+    let leftNav, rightNav, content, pagiNation;
```
Comment (nightpool) [inline app/javascript/mastodon/features/ui/components/media_modal.js]: should be pagination, here and throughout.

File: app/javascript/styles/components.scss [hunk 2]
```diff
@@ -3061,6 +3061,33 @@ button.icon-button.active i.fa-retweet {
...
+  text-align: center;
+  position: absolute;
+  left: 0;
+  bottom: -40px;
+
+  li {
+    display: inline-block;
+  }
+
+  button {
```
Comment (nightpool) [inline app/javascript/styles/components.scss]: you should generally avoid nested selectors in your css. 
instead, you should use an element class, like `.media-model__pagination_button` (or something less wordy :D)

**Review by nightpool — CHANGES_REQUESTED** [review]

**Comment by pfm-eyesightjp** [comment]
Thank you for your review.
I fixed.

File: app/javascript/mastodon/features/ui/components/media_modal.js [hunk 3]
```diff
@@ -67,10 +72,17 @@ export default class MediaModal extends ImmutablePureComponent {
...
 
     const index = this.getIndex();
+    const pagination = [];
 
     const leftNav  = media.size > 1 && <button tabIndex='0' className='modal-container__nav modal-container__nav--left' onClick={this.handlePrevClick} aria-label={intl.formatMessage(messages.previous)}><i className='fa fa-fw fa-chevron-left' /></button>;
     const rightNav = media.size > 1 && <button tabIndex='0' className='modal-container__nav  modal-container__nav--right' onClick={this.handleNextClick} aria-label={intl.formatMessage(messages.next)}><i className='fa fa-fw fa-chevron-right' /></button>;
 
+    if (media.size > 1) {
+      media.forEach((item, i) => {
+        pagination.push(<li className='media-modal__list' key={i}><button tabIndex='0' className={`media-modal__button${i === index ? ' active' : ''}`} onClick={this.handleChangeIndex} data-index={i}>{i + 1}</button></li>);
```
Comment (nightpool) [inline app/javascript/mastodon/features/ui/components/media_modal.js]: `.active` is still nested here. it should be `${i === index ? ' media-modal__button--active' : ''}`

furthermore, keeping the classname stuff inline is pretty hard to read. I would use `classes()` and pull out the entire class list to an intermediate variable.

File: app/javascript/styles/components.scss [hunk 4]
```diff
@@ -3083,6 +3083,33 @@ button.icon-button.active i.fa-retweet {
...
 }
 
+.media-modal__pagination {
+  width: 100%;
+  text-align: center;
+  position: absolute;
+  left: 0;
+  bottom: -40px;
+
+  .media-modal__list {
```
Comment (nightpool) [inline app/javascript/styles/components.scss]: these styles are still nested.

**Review by nightpool — DISMISSED** [review]

**Comment by nightpool** [comment]
Sorry for the late reply, I wasn't tracking this pull request.

**Comment by pfm-eyesightjp** [comment]
thanks.
I fixed again.

> Sorry for the late reply, I wasn't tracking this pull request. I'll try to keep up to date on it from now on.

OK, never mind!

File: app/javascript/mastodon/features/ui/components/media_modal.js [hunk 5]
```diff
@@ -67,10 +72,21 @@ export default class MediaModal extends ImmutablePureComponent {
     const { media, intl, onClose } = this.props;
 
     const index = this.getIndex();
+    const pagination = [];
 
     const leftNav  = media.size > 1 && <button tabIndex='0' className='modal-container__nav modal-container__nav--left' onClick={this.handlePrevClick} aria-label={intl.formatMessage(messages.previous)}><i className='fa fa-fw fa-chevron-left' /></button>;
     const rightNav = media.size > 1 && <button tabIndex='0' className='modal-container__nav  modal-container__nav--right' onClick={this.handleNextClick} aria-label={intl.formatMessage(messages.next)}><i className='fa fa-fw fa-chevron-right' /></button>;
 
+    if (media.size > 1) {
+      media.forEach((item, i) => {
```
Comment (akihikodaki) [inline app/javascript/mastodon/features/ui/components/media_modal.js]: What about using `media.map`?

**Review by akihikodaki — CHANGES_REQUESTED** [review]

**Comment by pfm-eyesightjp** [comment]
Thank you.
I fixed it.

File: app/javascript/mastodon/features/ui/components/media_modal.js [hunk 6]
```diff
@@ -40,6 +40,11 @@ export default class MediaModal extends ImmutablePureComponent {
     this.setState({ index: (this.props.media.size + this.getIndex() - 1) % this.props.media.size });
   }
 
+  handleChangeIndex = (e) => {
+    const index = Number(e.currentTarget.getAttribute('data-index'));
+    this.setState({ index: (index) % this.props.media.size });
```
Comment (akihikodaki) [inline app/javascript/mastodon/features/ui/components/media_modal.js]: * I do not think it needs parentheses around `index`.
* Is there any case `% this.props.media.size` matters?

**Review by akihikodaki — CHANGES_REQUESTED** [review]

File: app/javascript/mastodon/features/ui/components/media_modal.js (hunk 6 above)
Comment (pfm-eyesightjp) [reply app/javascript/mastodon/features/ui/components/media_modal.js]: I referred to the existing code.
https://github.com/tootsuite/mastodon/blob/master/app/javascript/mastodon/features/ui/components/media_modal.js#L32

I removed them.

**Review by akihikodaki — APPROVED** [review]
Thank you for answering my late requests. It is ready to merge.

**Pull request merged**